### PR TITLE
Top level gem azure_sdk bundling azure management and data plane

### PR DESCRIPTION
**Pull request by vishrutshah**
Top level gem azure_sdk bundling azure management and data plane.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vishrutshah__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by stuartpreston** [comment]
Wow this looks like an interesting commit...! - does this mean the individual resource gems will not be loaded to RubyGems in future? /cc @devigned

**Comment by devigned** [comment]
@stuartpreston I believe that the gems will still be released as they are, but we'll also introduce a roll up gem to bring them together and provide a simpler interface when using multiple clients.

@vishrutshah or @veronicagg please correct me if I've misspoke.

File: azure_sdk/lib/azure_sdk/client.rb [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+      @credentials
+    end
+
+    #
+    # Creates a new instance of the Azure::ARM::Authorization::AuthorizationManagementClient class if does not exists.
+    # @param base_url [String] the base URI of the service.
+    # @param options [Array] filters to be applied to the HTTP requests.
+    # @return [Azure::ARM::Authorization::AuthorizationManagementClient] Azure ARM Authorization Management Client.
+    #
+    def authorization(base_url = nil, options = nil)
```
Comment (devigned) [inline azure_sdk/lib/azure_sdk/client.rb]: a little bit of meta programing would go a long way here :)

Comment (vishrutshah) [inline azure_sdk/lib/azure_sdk/client.rb]: @devigned That's the killer idea. Ruby Meta programming rocks :) 

It'd appreciate if you can spare sometime to review this PR. Thanks!

**Comment by vishrutshah** [comment]
@veronicagg Feel free to review this PR as you get a chance. Thanks!

**Comment by devigned** [comment]
LGTM 👍

File: azure_sdk/lib/azure_sdk/client.rb [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    # Check if this client is configured with the same options.
+    # @param opts [Hash] hash of client options.
+    # @return [Boolean] True if client is configured with same set of options, False otherwise.
+    #
+    def same_options?(opts)
+      opts.hash == @options.hash
+    end
+
+    #
+    # Check if this client is configured with the some options.
```
Comment (veronicagg) [inline azure_sdk/lib/azure_sdk/client.rb]: should this description be updated?

File: azure_sdk/lib/azure_sdk/version.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# encoding: utf-8
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+
+module Azure
+    VERSION = '0.5.0' # File.read(File.expand_path('../../../../ARM_VERSION', __FILE__)).strip
```
Comment (veronicagg) [inline azure_sdk/lib/azure_sdk/version.rb]: should the comment be removed?

File: azure_sdk/spec/azure_sdk_spec.rb [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    end
+
+    it 'should initialize with client constructor' do
+      arm_client = Azure::ARM.client(@minimum_valid_options)
+      verify_client_defaults(arm_client)
+      verify_management_clients(arm_client)
+    end
+
+    it 'should initialize by defaulting to ENV' do
+      arm_client = Azure::ARM.client({})
```
Comment (veronicagg) [inline azure_sdk/spec/azure_sdk_spec.rb]: do we need ({})?

File: azure_sdk/spec/azure_sdk_spec.rb [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+      expect(arm_client.credentials).to  be_kind_of(MsRest::ServiceClientCredentials)
+      verify_management_clients(arm_client)
+    end
+
+    it 'should initialize with credentials and subscription id' do
+      Azure::ARM.credentials = MsRest::TokenCredentials.new(MsRestAzure::ApplicationTokenProvider.new(
+          @tenant_id, @client_id, @client_secret, @active_directory_settings))
+      Azure::ARM.subscription_id = @subscription_id
+
+      verify_management_clients(Azure::ARM)
```
Comment (veronicagg) [inline azure_sdk/spec/azure_sdk_spec.rb]: should we be validating that credentials of at least one client matches what you set it to? and would that be enough validation for this test?

**Comment by veronicagg** [comment]
@vishrutshah Left minor comments, rest looks great to me, thanks!

File: azure_sdk/lib/azure_sdk/client.rb (hunk 2 above)
Comment (vishrutshah) [inline azure_sdk/lib/azure_sdk/client.rb]: Yup. Corrected.

---
In reply to: [76343363](https://github.com/Azure/azure-sdk-for-ruby/pull/433#discussion_r76343363) [](ancestors = 76343363)

File: azure_sdk/spec/azure_sdk_spec.rb (hunk 4 above)
Comment (vishrutshah) [inline azure_sdk/spec/azure_sdk_spec.rb]: nope. removed

---
In reply to: [76344399](https://github.com/Azure/azure-sdk-for-ruby/pull/433#discussion_r76344399) [](ancestors = 76344399)

File: azure_sdk/lib/azure_sdk/version.rb (hunk 3 above)
Comment (vishrutshah) [inline azure_sdk/lib/azure_sdk/version.rb]: Yup not required.

---
In reply to: [76344267](https://github.com/Azure/azure-sdk-for-ruby/pull/433#discussion_r76344267) [](ancestors = 76344267)

File: azure_sdk/spec/azure_sdk_spec.rb (hunk 5 above)
Comment (vishrutshah) [inline azure_sdk/spec/azure_sdk_spec.rb]: Sure

---
In reply to: [76344795](https://github.com/Azure/azure-sdk-for-ruby/pull/433#discussion_r76344795) [](ancestors = 76344795)

**Comment by vishrutshah** [comment]
@veronicagg Thanks for the review! I've addressed the comments. Feel free to skim through it again. 

---
In reply to: [242583285](https://github.com/Azure/azure-sdk-for-ruby/pull/433#issuecomment-242583285) [](ancestors = 242583285)

**Pull request merged**